go/pkg/log/server/property_test.go

```diff
@@ -0,0 +1,158 @@
+		CollectionCompactionOffset: map[types.UniqueID]int64{},
+	}
+}
+
+func (suite *LogServerTestSuite) TestRecordLogDb_PushLogs() {
+	ctx := context.Background()
+	// Generate collection ids
+	collections := make([]types.UniqueID, 10)
+	for i := 0; i < len(collections); i++ {
+		collections[i] = types.NewUniqueID()
+	}
+
+	collectionGen := rapid.Custom(func(t *rapid.T) types.UniqueID {
+		return collections[rapid.IntRange(0, len(collections)-1).Draw(t, "collection_id")]
+	})
+	recordGen := rapid.SliceOf(rapid.Custom(func(t *rapid.T) *coordinatorpb.OperationRecord {
+		data := rapid.SliceOf(rapid.Byte()).Draw(t, "record_data")
+		id := rapid.String().Draw(t, "record_id")
+		return &coordinatorpb.OperationRecord{
+			Id: id,
+			Vector: &coordinatorpb.Vector{
+				Vector: data,
+			},
+		}
+	}))
+	rapid.Check(suite.t, func(t *rapid.T) {
+		t.Repeat(map[string]func(*rapid.T){
+			"pushLogs": func(t *rapid.T) {
+				c := collectionGen.Draw(t, "collection")
+				records := recordGen.Draw(t, "record")
+				r, err := suite.logServer.PushLogs(ctx, &logservicepb.PushLogsRequest{
+					CollectionId: c.String(),
+					Records:      records,
+				})
+				if err != nil {
+					t.Fatal(err)
+				}
+				if int32(len(records)) != r.RecordCount {
+					t.Fatal("record count mismatch", len(records), r.RecordCount)
+				}
+				suite.model.CollectionData[c] = append(suite.model.CollectionData[c], records...)
+			},
+			"getAllCollectionsToCompact": func(t *rapid.T) {
+				result, err := suite.logServer.GetAllCollectionInfoToCompact(ctx, &logservicepb.GetAllCollectionInfoToCompactRequest{})
+				assert.NoError(suite.t, err)
+				for _, collection := range result.AllCollectionInfo {
+					id, err := types.Parse(collection.CollectionId)
+					if err != nil {
+						t.Fatal(err)
+					}
+					compactionOffset := rapid.Int64Range(suite.model.CollectionCompactionOffset[id], int64(len(suite.model.CollectionData))).Draw(t, "new_position")
+					_, err = suite.logServer.UpdateCollectionLogOffset(ctx, &logservicepb.UpdateCollectionLogOffsetRequest{
+						CollectionId: id.String(),
+						LogOffset:    compactionOffset,
+					})
+					if err != nil {
+						t.Fatal(err)
+					}
+					suite.model.CollectionCompactionOffset[id] = compactionOffset
+				}
+			},
+			"pullLogs": func(t *rapid.T) {
+				c := collectionGen.Draw(t, "collection")
+				startOffset := rapid.Int64Range(suite.model.CollectionCompactionOffset[c], int64(len(suite.model.CollectionData))).Draw(t, "start_offset")
+				// If start offset is 0, we need to set it to 1 as the offset is 1 based
+				if startOffset == 0 {
+					startOffset = 1
+				}
+				batchSize := rapid.Int32Range(1, 20).Draw(t, "batch_size")
+				response, err := suite.logServer.PullLogs(ctx, &logservicepb.PullLogsRequest{
+					CollectionId:    c.String(),
+					StartFromOffset: startOffset,
+					BatchSize:       batchSize,
+					EndTimestamp:    time.Now().Unix(),
```

Comment:
Can we also generate the EndTimestamp?

Reply:
yes, once I have it deploy to staging, i would like to pass even more time on making the property test even better, like parallelism, more logs, use timestamp etc